enhancements/monitoring/monitoring-config.md

```diff
@@ -0,0 +1,190 @@
+---
+title: Configuration CRD for the Cluster Monitoring Operator
+authors:
+  - "@marioferh"
+  - "@danielmellado"
+  - "@jan--f"
+  - "@moadz"
+  - "simonpasquier"
+reviewers:
+  - "@jan--f"
+  - "@moadz"
+  - "simonpasquier"
+approvers:
+  - "@jan--f"
+  - "@moadz"
+  - "simonpasquier"
+  - "@openshift/openshift-team-monitoring"
+creation-date: 2024-04-26
+last-updated: 2024-04-26
+status: provisional
+---
+
+# CRD ClusterMonitoring
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+* Currently, the monitoring stack is configured using a configmaps. In OpenShift though the best practice is to configure operators using custom resources.
+
+
+## Motivation
+
+* The specification is well known and to a degree self-documenting
+* The APIServer will validate user resources based on our specifications, so users get immediate feedback on errors instead of having to check if their config was applied and check logs.
+* Many users expect to interact with operators through a CRD
+* Compatible with GitOps workflows. 
+* We can add [cross]validation rules to CRD fields to avoid misconfigurations
+* End users get a much faster feedback loop. No more applying the config and scanning logs if things don't look right. The API server will give immediate feedback
+* Organizational users (such as ACM) can manage a single resource and observe its status
+
+### Goals
+
+- Replace configmaps with CRD
+- Smooth transition for users
+
+## Proposal
+
+### Overview
+
+- Replace confimgaps with CRD:
```

Comment:
Can you add a bit more details on the current implementation and the overall context for readers not familiar with the product?

* Describe the 2 stacks (platform & UWM)
* How does an admin configure each stack? Relationship between the 2?
* Explain that there can be 2 personas responsible for each stack (the managed OpenShift use case). 

A link to https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/multi-tenant-alerting.md would be useful.

Reply:
Add improvements